File: app/src/main/java/com/example/weatherapp/api/ApiInterface.java
package com.example.weatherapp.api;

import com.example.weatherapp.models.citywiseweather.WeatherDetails;
import com.example.weatherapp.models.latlngwiseweather.CurLocationWeatherDetails;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ApiInterface {

    @GET("/data/2.5/weather")
    Call<WeatherDetails> cityWeatherDetails(@Query("q") String cityName, @Query("appid") String appId);

    @GET("/data/2.5/weather")
    Call<CurLocationWeatherDetails> curLocationWeatherDetails(@Query("lat") String lat, @Query("lon") String lon, @Query("appid") String appId);



}

File: app/src/main/java/com/example/weatherapp/api/ApiInterface.kt
package com.example.weatherapp.api

import com.example.weatherapp.model.current_weather_data.CurrentWeatherData
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Query

interface ApiInterface {

    @GET("/data/2.5/weather")
    suspend fun getCurrentWeatherData(
        @Query("lat") latitude: Double,
        @Query("lon") longitude: Double,
        @Query("appid") apiKey: String
    ) : Response<CurrentWeatherData>

}